src/main/kotlin/com/openlattice/transporter/TransporterQueries.kt

```diff
@@ -0,0 +1,145 @@
+package com.openlattice.transporter
+
+import com.openlattice.ApiUtil
+import com.openlattice.IdConstants
+import com.openlattice.postgres.*
+import com.openlattice.postgres.PostgresColumn.*
+import com.openlattice.transporter.types.TransporterColumn
+import com.zaxxer.hikari.HikariDataSource
+import java.util.*
+
+private val transportTimestampColumn: PostgresColumnDefinition = LAST_PROPAGATE
+
+fun unquotedTableName(entityTypeId: UUID): String {
+    return "et_$entityTypeId"
+}
+
+fun tableName(entityTypeId: UUID): String {
+    return ApiUtil.dbQuote(unquotedTableName(entityTypeId))
+}
+
+fun tableDefinition(entityTypeId: UUID, propertyColumns: Collection<PostgresColumnDefinition>): PostgresTableDefinition {
+    val definition = PostgresTableDefinition(tableName(entityTypeId))
+    val indexPrefix = unquotedTableName(entityTypeId) + "_"
+    definition.addColumns(
+            ENTITY_SET_ID,
+            ID_VALUE,
+            ORIGIN_ID
+    )
+    definition.primaryKey(
+            ENTITY_SET_ID,
+            ID_VALUE
+    )
+    definition.addIndexes(
+            PostgresExpressionIndexDefinition(definition, "(${ORIGIN_ID.name} != '${IdConstants.EMPTY_ORIGIN_ID.id}')" )
+                    .name(ApiUtil.dbQuote(indexPrefix + "origin_id"))
+                    .ifNotExists()
+                    .concurrent()
+    )
+    definition.addColumns(*propertyColumns.toTypedArray())
+    return definition
+}
+
+/**
+ * column bindings are
+ * 1 - partitions array
+ * 2 - entity set ids array
+ */
+private val checkQuery = "SELECT 1 " +
+        "WHERE EXISTS (" +
+        " SELECT 1 FROM ${PostgresTable.IDS.name} " +
+        " WHERE ${PARTITION.name} = ANY(?) " +
+        "  AND ${ENTITY_SET_ID.name} = ANY(?) " +
+        "  AND ${DataTables.LAST_WRITE.name} > ${transportTimestampColumn.name}" +
+        ")"
+
+fun hasModifiedData(enterprise: HikariDataSource, partitions: Collection<Int>, entitySetIds: Set<UUID>): Boolean {
+    return enterprise.connection.use { conn ->
+        conn.prepareStatement(checkQuery).use { st ->
+            val partitionsArray = PostgresArrays.createIntArray(conn, partitions)
+            val entitySetArray = PostgresArrays.createUuidArray(conn, entitySetIds)
+            st.setArray(1, partitionsArray)
+            st.setArray(2, entitySetArray)
+            val rs = st.executeQuery()
+            rs.next()
+        }
+    }
+}
+
+val pkCols = listOf(ENTITY_SET_ID, ID_VALUE)
+val pk = pkCols.joinToString(", ") { it.name }
+val ids = "$pk,${ORIGIN_ID.name}"
+
+/**
+ * column bindings are
+ * 1 - partitions array
+ * 2 - entity set ids array
+ *
+ */
+fun updateOneBatchForProperty(destTable: String, propId: UUID, propCol: TransporterColumn): String {
+    val srcCol = propCol.srcCol
+    val destCol = propCol.destColName
+    val dataCol = "CASE WHEN VERSION > 0 THEN $srcCol else null END as $destCol"
```

Comment:
What do `srcCol` and `destCol` map to? And why is the decision about which contains data affected by `version`?

Reply:
srcCol is the result of `PostgresDataTables.getColumnDefinition()` on the property, and destCol is the uuid of the property, but quoted. The constructor for TransporterColumn isn't well documented, I'll update that (and rename the variables to something a little more clear).